python/ray/serve/pipeline/api.py

```diff
@@ -67,6 +69,19 @@ def build(ray_dag_root_node: DAGNode) -> List[Deployment]:
             lambda node: transform_ray_dag_to_serve_dag(node, node_name_generator)
```

Comment:
now i'm wondering, can we directly go from ClassNode/FunctionNode to these ExecutorNode? The transform process can spit out:
- A list of deployment objects
- A minimal dag for execution

So we can delete all the existing Deployment{, Function, Method}Node implementations

Reply:
yes this is exactly my thinking. I think we only need to do the following:

- Collect list of deployments that need to be deployed (w/ args, options).
- Transform ClassNode -> ServeHandle.
- Transform {ClassMethodNode, FunctionNode} -> DAGExecutorNode.

Reply:
we eventually should be able to, and in April i spent about a week trying it but it has a bit more magic regarding how we deal with deployment and dag handles as deployment arg in our 1.12 sprint .. so this PR is more about adding the final ideal state with perf wins, then we can work backwards to simplify what happened in first pass.